Local-mode: Renaming to app from application (#2015)
balena-io/docs: pages/learn/develop/local-mode.md

Local mode also has another huge benefit, known as [Livepush][livepush]. Livepush makes intelligent decisions on how, or even if, to rebuild an image when changes are made. Instead of creating a new image and container with every code change, the Dockerfile commands are executed from within the running container. This means that, for example, if you added a dependency to your `package.json`, rather than having to install all of the dependencies again, only the new dependency would be installed.

> Once a file has been modified in the app, the Supervisor will immediately detect the change and then either rebuild the image or, for source files that run in-service, replace the changed files in situ in the relevant container layer and restart the service. As this happens in a few seconds, it makes the process of developing much faster and more convenient.

Reviewer: ```suggestion
When a source file is modified, the Supervisor will immediately detect the change and then either rebuild the image or, for source files that run in-service, replace the changed files in situ in the relevant container layer and restart the service. As this happens in a few seconds, it makes the process of developing much faster and more convenient.
```

```bash
[Live]    Detected changes for container main, updating...
[Live]    [main] Restarting service..
```

> __Note:__ You can disable Livepush by passing the `--nolive` option to `{{ $names.company.short }} push`. To rebuild the app on the device you will need to perform another `{{ $names.company.short }} push`.

Reviewer: ```suggestion
__Note:__ You can disable Livepush by passing the `--nolive` option to `{{ $names.company.short }} push`. In this case to rebuild on the device you will need to perform another `{{ $names.company.short }} push`.
```

{{ $names.company.short }} ssh 192.168.86.45
```

> To connect to an app container, we can specify the service name e.g.

Reviewer: ```suggestion
To connect to a container, we can specify the service name e.g.
```

```bash
sudo {{ $names.company.short }} ssh 63ec46c.local my-service
```

> __Note:__ If an IP address or a `.local` hostname is used (instead of an app name or device UUID), `{{ $names.company.short }} ssh` establishes a direct connection to the device on port `22222` that does not rely on the {{ $names.company.short }} VPN.

Reviewer: ```suggestion
__Note:__ If an IP address or a `.local` hostname is used (instead of a fleet name or device UUID), `{{ $names.company.short }} ssh` establishes a direct connection to the device on port `22222` that does not rely on the {{ $names.company.short }} VPN.
```
